councilmatic_core/models.py

```diff
@@ -309,8 +309,16 @@ class Meta:
 
 class EventManager(models.Manager):
     def get_queryset(self):
-        return super().get_queryset().annotate(start_time=Cast('start_date',
-                                                               models.DateTimeField()))
+        return super().get_queryset().annotate(
+            start_time=Cast(
+                Case(
```

Comment:
we use this same annotation in a handful of places, and it seems like we may want to use it more in the future, if we decide to handle all datetimes. what if we defined a mixin with a helper method and included it in the custom managers that need this functionality?

```python
class CastToDateTimeMixin:
    def cast_to_datetime(self, field):
        return Cast(
            Case(
                When(**{field: '', 'then': None}),
                default=field,
                output_field=models.CharField()
            ),
            models.DateTimeField()
        )

class EventManager(CastToDateTimeMixin, models.Manager):
    def get_queryset(self):
        return super().get_queryset().annotate(
            start_time=self.cast_to_datetime('start_date')
        )
```

Reply:
This sounds like a great idea! I'll make the change. What's the advantage of making it a mixin as opposed to a standalone function? Seems like it doesn't actually interface with the `Manager` methods or properties at all.